suite-native/module-trading/src/components/general/TradeableAssetAccountBalance.tsx

```diff
@@ -0,0 +1,104 @@
+import { useSelector } from 'react-redux';
+
+import { BASE_CRYPTO_MAX_DISPLAYED_DECIMALS } from '@suite-common/formatters';
+import { isSendingEvmNativeToken, parseCryptoId } from '@suite-common/trading';
+import { NetworkSymbol } from '@suite-common/wallet-config';
+import { Account, TokenAddress, TokenSymbol } from '@suite-common/wallet-types';
+import { DiscreetTextTrigger, HStack, Text } from '@suite-native/atoms';
+import { CryptoAmountFormatter, TokenAmountFormatter } from '@suite-native/formatters';
+import { Translation } from '@suite-native/intl';
+import { TokensRootState, selectAccountTokenBalance } from '@suite-native/tokens';
+
+import { TradeableAsset } from '../../types/general';
+
+export type TradeableAssetAccountBalanceProps = {
+    account?: Account;
+    asset?: TradeableAsset;
+    testID?: string;
+};
+
+type TokenBalanceProps = {
+    accountKey: Account['key'];
+    tokenAddress: TokenAddress;
+    symbol: string;
+    testID?: string;
+};
+
+type AssetBalanceProps = {
+    account: Account;
+    asset: TradeableAsset;
+    testID?: string;
+};
+
+const TokenBalance = ({ accountKey, tokenAddress, symbol, testID }: TokenBalanceProps) => {
+    const balance = useSelector(
+        (state: TokensRootState) =>
+            selectAccountTokenBalance(state, accountKey, tokenAddress) ?? '0',
+    );
+
+    return (
+        <TokenAmountFormatter
+            value={balance}
+            tokenSymbol={symbol.toUpperCase() as TokenSymbol}
+            testID={testID}
+        />
+    );
+};
+
+const AssetBalance = ({ account, asset, testID }: AssetBalanceProps) => {
+    const { cryptoId, symbol } = asset;
+    const { key: accountKey, formattedBalance } = account;
+
+    const tokenAddress = isSendingEvmNativeToken(cryptoId)
+        ? undefined
+        : parseCryptoId(cryptoId).contractAddress;
+
+    return (
+        <DiscreetTextTrigger>
+            {tokenAddress ? (
+                <TokenBalance
+                    symbol={symbol}
+                    accountKey={accountKey}
+                    tokenAddress={tokenAddress as TokenAddress}
```

Comment:
if asset.contractAddress is used, this is not needed anymore

```suggestion
                    tokenAddress={tokenAddress}
```